### PR TITLE
[Issue 3649] Altair accounting reform - reward and penalty calculations

### PR DESCRIPTION
**Pull request by mbaxter**
## PR Description
Implement Altair reward and penalty calculations (see: `RewardsAndPenaltiesCalculatorAltair`).

Other changes include:
* Move `TotalBalances` into `spec` package
* Update `ValidatorStatus` and `TotalBalances` variables / getters to be more descriptive
* Rename `Deltas` to `RewardAndPenaltyDeltas` for clarity
* Update some constants / config types
* Add missing getters to `BeaconStateAltair`

## Fixed Issue(s)
Part of #3649 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java [hunk 1]
```diff
@@ -41,29 +43,33 @@
...
     this.specConfig = specConfig;
+    this.miscHelpers = miscHelpers;
     this.validatorsUtil = validatorsUtil;
     this.beaconStateUtil = beaconStateUtil;
     this.validatorStatusFactory = validatorStatusFactory;
     this.beaconStateAccessors = beaconStateAccessors;
   }
 
-  private static void applyDeltas(final MutableBeaconState state, final Deltas attestationDeltas) {
+  protected static void applyDeltas(
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java]: Seems weird that this is static and I don't think it needs to be.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/RewardsAndPenaltiesCalculatorAltair.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+    return deltas;
+  }
+
+  /**
+   * Corresponds to altair beacon chain accessor get_flag_index_deltas
+   *
+   * @see <a
+   *     href="https://github.com/ethereum/eth2.0-specs/blob/master/specs/altair/beacon-chain.md#beacon-state-accessors">Altair
+   *     beacon-chain.md</a>
```
Comment (ajsutton) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/RewardsAndPenaltiesCalculatorAltair.java:81]: These links have tended to not age well. I'd be tempted to omit it.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/RewardsAndPenaltiesCalculatorAltair.java (hunk 2 above)
Comment (mbaxter) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/statetransition/epoch/RewardsAndPenaltiesCalculatorAltair.java:81]: True, although I think the spec is much more stable than it used to be.  And now that we're live, things can't just change arbitrarily.  It was a little hard to map some of the phase0 logic to the spec, so I think it does help having these notes when we're working on updates.

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java (hunk 1 above)
Comment (mbaxter) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java]: 👍 updated

**Pull request merged**